test/cmds/watch/watch.test.js

```diff
@@ -0,0 +1,148 @@
+const EventEmitter = require('events');
+const chokidar = require('chokidar');
+const watch = require('../../../dist/src/cmds/watch');
+const build = require('../../../dist/src/cmds/build/bundle');
+const fsUtils = require('../../../dist/src/utils/validate-fs');
+const miscUtils = require('../../../dist/src/utils/misc');
+
+describe('watch', () => {
+  describe('Watch a directory and its subdirectories for changes, and build when files are added or changed.', () => {
+
+    let watcherEmitter;
+
+    const mockArgv = {
+      src: 'index.js',
+      dist: 'dist',
+      outfileName: 'bundle.js',
+    };
+
+    beforeEach(() => {
+      jest.spyOn(chokidar, 'watch').mockImplementation(() => {
+        watcherEmitter = new EventEmitter();
+        watcherEmitter.add = () => undefined;
+        jest.spyOn(watcherEmitter, 'on');
+        jest.spyOn(watcherEmitter, 'add');
+        return watcherEmitter;
+      });
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+      jest.restoreAllMocks();
+      watcherEmitter = undefined;
+    });
+
+    it('successfully processes arguments from yargs', async () => {
+      jest.spyOn(console, 'log').mockImplementation();
+      const validateDirPathMock = jest.spyOn(fsUtils, 'validateDirPath').mockImplementation(() => true);
+      const validateFilePathMock = jest.spyOn(fsUtils, 'validateFilePath').mockImplementation(() => true);
+      const validateOutfileNameMock = jest.spyOn(fsUtils, 'validateOutfileName').mockImplementation(() => true);
+      const getOutfilePathMock = jest.spyOn(fsUtils, 'getOutfilePath').mockImplementation(() => 'dist/bundle.js');
+      await watch.handler(mockArgv);
+      expect(validateDirPathMock).toHaveBeenCalledTimes(1);
+      expect(validateFilePathMock).toHaveBeenCalledTimes(1);
+      expect(validateOutfileNameMock).toHaveBeenCalledTimes(1);
+      expect(getOutfilePathMock).toHaveBeenCalledTimes(1);
```

Comment:
I think it's worth checking that validation functions have been called in this test case, because we want at least one check that we validate the input to the handler.

However, I think we should remove the check for whether `getOutfilePathMock` is called, and instead confirm that `chokidar.watch` was called with the expected path string as its first argument (the variable `root` in the file).